### ISSUE TITLE
[Bug]: Edit location button shown even when image does not have location

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#5908]
### Summary

When trying to upload an image, there seems to be some kind of uncertainity with which the app identifies the presence of location in the image. In an instance, the "Edit location" button was shown for an image that did not have location in the EXIF. The "No location found" prompt is properly shown, though. This is in a device running Android 12.

Note that I'm not able to reliably reproduce this for all images. I do see the "Add location" button for other images. For example:

- [This is an image](https://drive.google.com/file/d/1xIIPBBepDk4s3Tpz7qt9OMXbpoHwxl4N/view?usp=sharing) for which I see the "Edit location" button incorrectly
- [This is an image](https://drive.google.com/file/d/1l7rLNdmpNEjTK_Ljie6afVrYQzqgfS2W/view?usp=sharing) for which the "Add location" button is shown correctly

### Steps to reproduce

1. Open the app
2. Try to upload the image found [here](https://drive.google.com/file/d/1xIIPBBepDk4s3Tpz7qt9OMXbpoHwxl4N/view?usp=sharing)
3. Notice the "Edit location" button shown
4. Fill in the "caption" and "description"
5. Click "Next" and notice the "No location found" prompt shown

### Expected behaviour

The app identifies that the image does not have location and shows the "Add location" button. 

### Actual behaviour

The app shows the "Edit location" button incorrectly for an image that does not have a location.

### Device name

OnePlus Nord

### Android version

Android 12

### Commons app version

5.1.0~522f1fe19

### Screen-shots

![Screenshot_2024-10-08-20-09-51-36_d5db3f3edc380047609fe9c266f7c566_1](https://github.com/user-attachments/assets/3c6e2689-8b53-426a-bec9-1e5ed0c61112)

### Would you like to work on the issue?

None

**Comment by nicolas-raoul** [comment]
> "[This is an image](https://drive.google.com/file/d/1xIIPBBepDk4s3Tpz7qt9OMXbpoHwxl4N/view?usp=sharing) for which I see the "Edit location" button incorrectly"

Reproducible every time?

**Comment by sivaraam** [comment]
With that image, yes I can reproduce it every time. Could you possibly try using those images and see what you observe?

**Comment by savsch** [comment]
I reproduced the issue and was able to pinpoint the cause.
> [This is an image](https://drive.google.com/file/d/1xIIPBBepDk4s3Tpz7qt9OMXbpoHwxl4N/view?usp=sharing) for which I see the "Edit location" button incorrectly

For this image (unlike the other), the location attribute does technically exist in the exif metadata, although it is malformed (as is evident from various tools: I tried exiftool and python's pillow).
In fact, `androidx.exifinterface` issues a warning in the logcat: _Latitude/longitude values are not parsable. latValue=0/1,0/1,0/1, latRef=, lngValue=0/1,0/1,0/1, lngRef=_ when `androidx.exifinterface.media.ExifInterface`'s `getAttribute(ExifInterface.TAG_GPS_LATITUDE)` method is called for that image.

Therefore, it returns a valid latValue and lngValue but an empty latRef (and lngRef) String.

When determining whether to show the "Edit Location" or the "Add Location" button, only the latValue (and lngValue) are used (UploadableFile.kt), but when determining whether to show the "No location found" prompt, the lngRef is also used (ImageCoordinates.kt) by the application.

I can help resolve this issue if I am assigned to it. @sivaraam @nicolas-raoul
